src/helpers.js

```diff
@@ -195,6 +204,230 @@ module.exports = ({ types: t }) => {
+    if (!t.isArrayExpression(dependencyList)) {
+      if (isDefineCall) {
+        // If the arity of the define call is 2, then the dependency list argument
+        // could be the module name if the type is string.  In that case, we ignore
+        // the module name and use the default dependencies per the spec:
+        // https://github.com/amdjs/amdjs-api/wiki/AMD#dependencies-
+        // Generated code:
+        // if (typeof amdDeps === 'string') {
+        //   amdDeps = ['require', 'exports', 'module'];
+        // }
+        if (arity === 2) {
+          blockStatements.push(
+            t.ifStatement(
+              t.binaryExpression(
+                '===',
+                t.unaryExpression('typeof', depsIdentifier),
+                t.stringLiteral('string')
+              ),
+              t.blockStatement([
+                t.expressionStatement(
+                  t.assignmentExpression(
+                    '=',
+                    depsIdentifier,
+                    t.arrayExpression([
+                      t.stringLiteral(REQUIRE),
+                      t.stringLiteral(EXPORTS),
+                      t.stringLiteral(MODULE),
+                    ])
+                  )
+                ),
+              ])
+            )
+          );
+        }
+      } else {
+        // Generated code:
+        // if (amdDeps === null || typeof amdDeps !== 'object' || isNaN(amdDeps.length)) {
+        //   return require(amdDeps);
+        // }
+        blockStatements.push(
+          t.ifStatement(
+            t.logicalExpression(
+              '||',
+              t.binaryExpression('===', depsIdentifier, t.nullLiteral()),
+              t.logicalExpression(
+                '||',
+                t.binaryExpression(
+                  '!==',
+                  t.unaryExpression('typeof', depsIdentifier),
+                  t.stringLiteral('object')
+                ),
+                t.callExpression(t.identifier('isNaN'), [
+                  t.memberExpression(depsIdentifier, t.identifier('length')),
+                ])
+              )
+            ),
+            t.blockStatement([
+              t.returnStatement(t.callExpression(t.identifier(REQUIRE), [depsIdentifier])),
+            ])
+          )
+        );
+      }
+      // Convert array-like objects to real arrays
+      // amdDeps = [].slice.call(amdDeps);
+      blockStatements.push(
+        t.expressionStatement(
+          t.assignmentExpression(
+            '=',
+            depsIdentifier,
+            t.callExpression(
+              t.memberExpression(
+                t.memberExpression(t.arrayExpression(), t.identifier('slice')),
+                t.identifier('call')
+              ),
+              [depsIdentifier]
+            )
+          )
+        )
+      );
+    }
+
+    // If we don't know that the factory is a function, then we need to check the type
+    // at runtime.
+    if (!isFunctionExpression(factory)) {
```

Comment:
This code path is now executed only when the dependency list is also not an array type.  so it makes more sense to use the new code here to check the factory type rather than try to reuse the existing code.